cpp/CMakeLists.txt

```diff
@@ -83,31 +83,44 @@ include(cmake/thirdparty/get_nvtx.cmake)
 # ##################################################################################################
 # * library targets --------------------------------------------------------------------------------
 
-add_library(rmm INTERFACE)
+add_library(rmm SHARED src/aligned.cpp src/cuda_device.cpp src/cuda_stream_pool.cpp
+                       src/cuda_stream_view.cpp src/cuda_stream.cpp)
 add_library(rmm::rmm ALIAS rmm)
 
 target_include_directories(
   rmm
-  INTERFACE "$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>"
-            "$<BUILD_INTERFACE:${CMAKE_CURRENT_BINARY_DIR}/include>" "$<INSTALL_INTERFACE:include>")
+  PUBLIC "$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>"
+         "$<BUILD_INTERFACE:${CMAKE_CURRENT_BINARY_DIR}/include>"
+  INTERFACE "$<INSTALL_INTERFACE:include>")
 
 if(CUDA_STATIC_RUNTIME)
```

Comment:
I would love to rework this so that rmm doesn't impart a hard requirement on the type of CUDART linking of downstream projects.

I think we can defer that to a future PR but we should make sure we have an issue for the work.

Reply:
I think https://github.com/rapidsai/rmm/issues/1320 is tracking that, if I understand correctly. I haven't devoted any time to that, and I am not really sure if I understand what to do.

Reply:
:+1:  That issue is sufficient

Reply:
Already exists, this is #1320